frontend/modules/viewerGui/viewerGui.sagas.ts

```diff
@@ -51,6 +51,19 @@ import {
 } from './viewerGui.selectors';
 
 function* fetchData({ teamspace, model }) {
+	try {
+		const {data: metaKeys} = yield API.getMetaKeys(teamspace, model);
+		yield put(ModelActions.fetchMetaKeysSuccess(metaKeys));
+	} catch (e) {
+		if (e?.response.status === 401) {
+			yield put(DialogActions.showEndpointErrorDialog('access', 'model', e));
```

Comment:
Actually if model settings failed, there's pretty much no point carrying on, so i don't think there's a need to reroute.

If we can make the dialog to be generic to just say "There's a problem accessing the model" , then show the network request error, and a button to redirect to teamspace list, that should do.

Reply:
This comment is marked as `Outdated` as I change code here, altho the general logic stays the same, this is why I'm not sure if it fits right now with Your comment or not. In general, if this error condition is met, no further action is continued.

Reply:
the comment is still applicable. I'll mark this as resolve and pin point it in the new version